verilog/dv/cocotb/hyperram_test/hyperram_test.c

```diff
@@ -0,0 +1,116 @@
+    // mux registers after 2 mux_conf_clk rising edges:
+    uint32_t la1 =
+     0b01111111101001010000000000000000;
+    // 0-------------------------------     mux_conf_clk: Start with mux configuration clock low
+    // -11111111-----------------------     i_design_reset[7:0]: All asserted
+    // ---------0----------------------     i_mux_auto_reset_enb: 0=auto-reset non-selected designs
+    // ----------1---------------------     i_mux_sys_reset_enb: 1=do not use wb_rst_i as a reset
+    // -----------0010-----------------     i_mux_sel[3:0]=1101: We'll select design 2 (HyperRAM)
+    // ---------------1----------------     i_mux_io5_reset_enb: 1=Do not use io[5] as a reset
+    // ----------------XXXXXXXXXXXXXXXX     Unused.
+    LogicAnalyzer_write(LA_REG_1, la1);
+
+    // Pulse mux_conf_clk once...
+    LogicAnalyzer_write(LA_REG_1, la1 |= 0x80000000);
+    LogicAnalyzer_write(LA_REG_1, la1 ^= 0x80000000);
+    // ...and again:
+    LogicAnalyzer_write(LA_REG_1, la1 |= 0x80000000);
+    LogicAnalyzer_write(LA_REG_1, la1 ^= 0x80000000);
+
+    // Let's set GPIO[37:8] to BIDIRECTIONAL mode, since our OEBs should
+    // be driven by the mux...
+    GPIOs_configureAll(GPIO_MODE_USER_STD_BIDIRECTIONAL); // Set all...
+    // ...but then change our mind for just the first 8:
+    GPIOs_configure(0, GPIO_MODE_MGMT_STD_INPUT_NOPULL);
+    GPIOs_configure(1, GPIO_MODE_MGMT_STD_INPUT_NOPULL);
+    GPIOs_configure(2, GPIO_MODE_MGMT_STD_INPUT_NOPULL);
+    GPIOs_configure(3, GPIO_MODE_MGMT_STD_INPUT_NOPULL);
+    GPIOs_configure(4, GPIO_MODE_MGMT_STD_INPUT_NOPULL);
+    GPIOs_configure(5, GPIO_MODE_MGMT_STD_INPUT_NOPULL);
+    GPIOs_configure(6, GPIO_MODE_MGMT_STD_INPUT_NOPULL);
+    GPIOs_configure(7, GPIO_MODE_MGMT_STD_INPUT_NOPULL);
+
+    // Design 2 (HyperRAM controller) is using GPIO[25-37]
+    GPIOs_configure(25, GPIO_MODE_USER_STD_OUTPUT);         // nRST
+    GPIOs_configure(26, GPIO_MODE_USER_STD_OUTPUT);         // nCS
+    GPIOs_configure(27, GPIO_MODE_USER_STD_OUTPUT);         // CLK
+    GPIOs_configure(28, GPIO_MODE_USER_STD_OUTPUT);         // nCLK
+    GPIOs_configure(29, GPIO_MODE_USER_STD_BIDIRECTIONAL);  // nRWDS
+    GPIOs_configure(30, GPIO_MODE_USER_STD_BIDIRECTIONAL);  // D0
+    GPIOs_configure(31, GPIO_MODE_USER_STD_BIDIRECTIONAL);
+    GPIOs_configure(32, GPIO_MODE_USER_STD_BIDIRECTIONAL);
+    GPIOs_configure(33, GPIO_MODE_USER_STD_BIDIRECTIONAL);
+    GPIOs_configure(34, GPIO_MODE_USER_STD_BIDIRECTIONAL);
+    GPIOs_configure(35, GPIO_MODE_USER_STD_BIDIRECTIONAL);
+    GPIOs_configure(36, GPIO_MODE_USER_STD_BIDIRECTIONAL);
+    GPIOs_configure(37, GPIO_MODE_USER_STD_BIDIRECTIONAL);  // D7
+
+    // Apply the above configuration:
+    GPIOs_loadConfigs();
+
+    // Pulse gpio again to show that the firmware configured GPIOS
+    pulse_gpio();       // 2nd pulse
+
+    // allow Design 2 to come out of reset
+    la1 &= ~(1ull << (57-32));
+    LogicAnalyzer_write(LA_REG_1, la1);
+    // Pulse mux_conf_clk once...
```

Comment:
The individual design reset lines are (hopefully!) ***un***registered, and take effect immediately, so after doing the write above in line 98, it's not necessary to do the `mux_conf_clk` pulses shown below, but it's also not a problem to do so :) If, however, you happen to find that your test doesn't work without lines 100 to 104, please let me know, because that might mean I've made a mistake :)